samza-core/src/main/java/org/apache/samza/execution/ProcessorGraph.java

```diff
@@ -307,13 +307,13 @@ private void validateReachability() {
       while (!q.isEmpty()) {
         ProcessorNode node = q.poll();
         sortedNodes.add(node);
+        visited.add(node);
```

Comment:
This handles a bug that this integration test harness found - In single node `ProcessorGraph`s, topological sort returns duplicate nodes. Please refer SAMZA-1172 for more context, and PR:#100 for the actual fix.